src/all/kavita/src/eu/kanade/tachiyomi/extension/all/kavita/KavitaHelper.kt

```diff
@@ -92,13 +92,13 @@ class KavitaHelper {
     fun chapterFromVolume(obj: ChapterDto, volume: VolumeDto): SChapter =
         SChapter.create().apply {
             // If there are multiple chapters to this volume, then prefix with Volume number
-            if (volume.chapters.isNotEmpty() && obj.number != "0") {
+            if (volume.chapters.isNotEmpty() && obj.number.toInt() > 0) {
                 // This volume is not volume 0, hence they are not loose chapters
                 // We just add a nice Volume X to the chapter title
                 // Chapter-based Volume
                 name = "Volume ${volume.number} Chapter ${obj.number}"
                 chapter_number = obj.number.toFloat()
-            } else if (obj.number == "0") {
+            } else if (obj.number.toInt() <= 0) {
                 // Both specials and volume has chapter number 0
                 if (volume.number == 0) {
```

Comment:
Wouldn't this need to check against the new encodings? 
Encoding: Special (100000), Loose Leaf Volume (-100000).

Specials are in Volume 100000 and loose leaf chapters are in -100000.